app/Helpers/database/player-game.php

```diff
@@ -115,10 +115,6 @@ function testBeatenGame(int $gameId, string $user, bool $postBeaten): array
                 displayOrder: 0
             );
         }
-
-        if (!RecentlyPostedProgressionActivity($user, $gameId, $awardMode, ActivityType::BeatGame)) {
-            postActivity($user, ActivityType::BeatGame, $gameId, $awardMode);
-        }
```

Comment:
It does seem weird that `postActivity` would update the `LastLogin` field, but it seems like it's being used as a "Last Activity" timestamp, so it's logically correct.
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/ece87c74-bbf6-49dc-8428-b800c9c6dfd9)

My understanding of the issue is that this is being called from some sort of background task to fill in missing information. In that case, either the activity needs to be back-dated, or just ignored.

In the normal use case, this seems like it should be expected. Maybe the `$postBeaten` check on line 105 should only apply to line 119, as the `HasSiteAward` should be sufficient to prevent doubling up on the SiteAward. Then the normal case would always pass true, and the recalculate case would pass false.

For now, removing this code seems reasonable. It can be properly addressed later.